src/Workspaces/Core/Portable/Utilities/IDictionaryExtensions.cs

```diff
@@ -21,14 +23,15 @@ internal static class IDictionaryExtensions
             return value;
         }
 
+        [return: MaybeNull]
         public static TValue GetValueOrDefault<TKey, TValue>(this IDictionary<TKey, TValue> dictionary, TKey key)
         {
             if (dictionary.TryGetValue(key, out var value))
             {
                 return value;
             }
 
-            return default;
+            return default!;
         }
 
         public static void MultiAdd<TKey, TValue, TCollection>(this IDictionary<TKey, TCollection> dictionary, TKey key, TValue value)
```

Comment:
Consider constraining TKey for all of these with a nonnull constraint.

Reply:
`IDictionary<TKey, TValue>` allows an implementation to support null keys.